### PR TITLE
MVC-form

### PR DESCRIPTION
**Pull request by MarinaPasternak**

### CODE REVIEW

File: app/components/acceptForm/form-controller.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { FormModel } from './form-model.js';
+
+export class FormController {
+  constructor({ subscribe}) {
+    this.view = new FormView(this.handleSubmitForm);
+    this.model = new FormModel();
+
+    this.subscribe = subscribe;
+
+    //use git pull to use events for publisherEvents
```
Comment (Anastasiia0999) [inline app/components/acceptForm/form-controller.js]: delete comments

**Review by rearchontis — CHANGES_REQUESTED** [review]
too much eslint errors

File: app/components/acceptForm/form-model.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+export class FormModel { 
+
+  getCardData = async (cardId) => {
+    const response = await fetch('http://www.json-generator.com/api/json/get/ctRVcUtcfC?indent=2');
```
Comment (Anastasiia0999) [inline app/components/acceptForm/form-model.js]: use property for url

File: app/components/acceptForm/form-view.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      let container = document.querySelector('.accept-inform');
+      let cardDetailsHtml = '';
+    
+     this.addDetails = (cardData) => {
+
+      cardDetailsHtml += `<p>Check that information is correct</p>
+            <p> ${cardData.model}</p>
+            <ul>`;
+
+      for (let key in cardData) {
```
Comment (Anastasiia0999) [inline app/components/acceptForm/form-view.js]: use separate function

File: app/components/acceptForm/form-controller.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  handleSubmitForm = (event) => {
+    event.preventDefault();
+    //this.view.clearErrors();
+
+    const { fullName, telephone } = this.model.getFormData(event);
+    this.notify(this.events.SEND_MASSAGE, { fullName, telephone } );
+
+    }
```
Comment (Anastasiia0999) [inline app/components/acceptForm/form-controller.js]: add ; after arrow func + delete 30 line

File: app/components/acceptForm/form-model.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    };
+
+      fetch(url, {
+        headers: {
+          "Content-Type": "application/json",
+        },
+        method: "POST",
+        body: JSON.stringify(payload),
+      });
+  }
```
Comment (Anastasiia0999) [inline app/components/acceptForm/form-model.js]: add ; here after arrow func

File: app/components/acceptForm/form-model.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        
+    };
+    
+  };
+
+  sendToBot = () =>{
+    const url = "https://kolimaga.herokuapp.com/";
+    const payload = {
+      payloadType: "confirm_order",
+      car,
```
Comment (rearchontis) [inline app/components/acceptForm/form-model.js]: Car is undefined, you possibly should use `getCardData` method

File: app/components/acceptForm/form-model.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      customer,
```
Comment (rearchontis) [inline app/components/acceptForm/form-model.js]: Customer is also undefined, you should use `getFormData`

File: app/components/acceptForm/form-model.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        if (el.type !== 'submit') {
+            dataObj[el.name] = el.value;
+        }
+        return dataObj;
+    }, {});
+   
+    return {
+        fullName: data['full-name'],
+        telephone: data['telephone'],
+
```
Comment (rearchontis) [inline app/components/acceptForm/form-model.js]: Should `email` field be here?

File: app/components/acceptForm/form-model.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    return {
+        fullName: data['full-name'],
+        telephone: data['telephone'],
+        
+    };
+    
+  };
+
+  sendToBot = () =>{
+    const url = "https://kolimaga.herokuapp.com/";
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-model.js]: Use single quotes, please

File: app/components/acceptForm/form-style.scss [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+.comfirm-form{
+    margin: auto;
+    border-radius: 10px;
+    display: grid;
+    height: fit-content;
+    padding: 20px;
+    background-color: rgb(209, 202, 230);
+} 
+
+input[type="text"]{
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-style.scss]: Better create class for input, using `input[type="text"]` you might overwrite input style in other components

File: app/components/acceptForm/form-view.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+                                   <input type="text" name="telephone" placeholder="Telephone">
+                                   <input type="submit" value="Accept">
+                                </form>
+                          </div>
+                      </div>`;
+    this.submitFormListener();
+  }
+
+    renderCarDetails = (cardData) => {
+      let container = document.querySelector('.accept-inform');
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: "container" variable wouldn't be overwritten, so you can use const.

File: app/components/acceptForm/form-view.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+      let cardDetailsHtml = '';
+    
+     this.addDetails = (cardData) => {
+
+      cardDetailsHtml += `<p>Check that information is correct</p>
+            <p> ${cardData.model}</p>
+            <ul>`;
+
+      for (let key in cardData) {
+        if (key != "photoUrl" && key != "id") {
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: Single quotes and !== instead of !=

File: app/components/acceptForm/form-view.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+                          </div>
+                      </div>`;
+    this.submitFormListener();
+  }
+
+    renderCarDetails = (cardData) => {
+      let container = document.querySelector('.accept-inform');
+      let cardDetailsHtml = '';
+    
+     this.addDetails = (cardData) => {
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: Why do you declare a method inside another method? Better move it to class

File: app/components/acceptForm/form-view.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+      for (let key in cardData) {
+        if (key != "photoUrl" && key != "id") {
+            cardDetailsHtml += `<li>${key}:  ${cardData[key]}</li>`;
+            }
+          }
+          cardDetailsHtml +='</ul>';
+        };
+
+        const carDetailsList = this.addDetails(cardData);
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: Unused constant

File: app/components/acceptForm/form-view.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
+import './form-style.scss';
+
+export class FormView {
+
+  main = document.querySelector('main');
+  
+  constructor(handleSubmitForm){
+    this.handleSubmitForm = handleSubmitForm
```
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: Add semicolon, please

File: app/components/acceptForm/form-view.js (hunk 3 above)
Comment (PavelKundenko) [inline app/components/acceptForm/form-view.js]: Use const instead of let

**Review by PavelKundenko — CHANGES_REQUESTED** [review]

File: app/components/acceptForm/form-controller.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+
+export class FormController {
+  constructor({ subscribe}) {
+    this.view = new FormView(this.handleSubmitForm);
+    this.model = new FormModel();
+
+    this.subscribe = subscribe;
+    //this.events = events;
+
+    this.subscribe('ADD_ORDER', this.handleAccept)
```
Comment (MozdolevskyiMaksym) [inline app/components/acceptForm/form-controller.js]: add this.events

**Review by rearchontis — CHANGES_REQUESTED** [review]
Too much eslint error.
Webpack server cannot start due to errors.

**Review by MozdolevskyiMaksym — CHANGES_REQUESTED** [review]

File: app/components/acceptForm/form-controller.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    const cardData = await this.model.getCardData(cardId);
+    this.view.renderContainer();
+    this.view.renderCarDetails(cardData);
+  };
+
+  handleSubmitForm = (event, cardId) => {
+    event.preventDefault();
+    //this.view.clearErrors();
+    const { fullName, telephone } = this.model.getFormData(event);
+    //this.notify(this.events.SEND_MASSAGE, { fullName, telephone } );
```
Comment (MozdolevskyiMaksym) [inline app/components/acceptForm/form-controller.js]: Delete comments

File: app/components/acceptForm/form-model.js [hunk 18]
```diff
@@ -0,0 +1,46 @@
+export class FormModel { 
+
+  getCardData = async (cardId) => {
+    const response = await fetch('http://www.json-generator.com/api/json/get/ctRVcUtcfC?indent=2');
+    const data = await response.json();
```
Comment (MozdolevskyiMaksym) [inline app/components/acceptForm/form-model.js]: why do you use fetch? You will get the car from Ivan`s component

File: app/components/acceptForm/form-view.js [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+       
+    };
+
+    addDetails = (cardData) => {
+      let cardDetailsHtml = '';
+      cardDetailsHtml += `<p>Check that information is correct</p>
+            <p> ${cardData.model}</p>
+            <ul>`;
+
+      for (const key in cardData) {
```
Comment (rearchontis) [inline app/components/acceptForm/form-view.js]: Please, don't use `for...in` method without necessary due to running through prototype, use arrays' methods and `Object.entries()`, `Object.keys()`, `Object.values()` instead, e.g. like in the [formatObjectToHTML method](https://github.com/so2niko/Dp-193_proj1/blob/dev/bot/helpers.js)

**Review by PavelKundenko — APPROVED** [review]

**Review by MozdolevskyiMaksym — APPROVED** [review]

File: app/main.js [hunk 20]
```diff
@@ -3,11 +3,12 @@ import Publisher from './helpers/publisher.js';
 import { ControllerCarsGallery } from './components/cars-gallery/controller-cars-gallery.js';
 import { FilterController } from './components/filter/filter-controller.js';
 import { HeaderController } from './components/header/header-controller.js';
+import { FormController } from './components/acceptForm/form-controller.js';
```
Comment (rearchontis) [inline app/main.js]: Change the path to `'./components/accept-form/form-controller.js'`

**Review by rearchontis — CHANGES_REQUESTED** [review]

File: app/main.js (hunk 20 above)
Comment (MarinaPasternak) [reply app/main.js]: done

**Review by rearchontis — APPROVED** [review]

**Review by Anastasiia0999 — APPROVED** [review]

**Pull request merged**